src/main/java/com/shopping/entities/Supplier.java

```diff
@@ -0,0 +1,46 @@
+package com.shopping.entities;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "supplier")
+public class Supplier extends BaseEntity{
+	
+	@Column
+	private String name;
+	
+	@OneToOne
+	private Address address;
+	
+	@OneToOne
+	private Product product;

```

Comment:
won't it be one to many since a supplier can supply multiple products?

Reply:
Then it will be many to many..
1 supplier can give n products
1 product can be given by many suppliers.

So shall i change to many to many

Reply:
Yes